pkg/apis/pravega/v1alpha1/status_test.go

```diff
@@ -73,12 +144,179 @@ var _ = Describe("PravegaCluster Status", func() {
+			})
+			It("Checking ClusterInRollbackFailedState should return false", func() {
+				Ω(p.Status.IsClusterInRollbackFailedState()).To(Equal(false))
+			})
+		})
+
+		Context("set pod upgrade condition to be false", func() {
+			BeforeEach(func() {
+				p.Status.SetUpgradingConditionTrue(" ", " ")
+				p.Status.SetUpgradingConditionFalse()
+			})
+
+			It("should have upgrade condition with false status", func() {
+				_, condition := p.Status.GetClusterCondition(v1alpha1.ClusterConditionUpgrading)
+				Ω(condition.Status).To(Equal(corev1.ConditionFalse))
+			})
+
+			It("should have upgrade condition with false status using function", func() {
+				Ω(p.Status.IsClusterInUpgradingState()).To(Equal(false))
+			})
+
+			It("Checking GetlastCondition function and It should return nil as not in Upgrading state", func() {
+				condition := p.Status.GetLastCondition()
+				Ω(condition).To(BeNil())
+			})
+
+			It("should have updated timestamps", func() {
+				_, condition := p.Status.GetClusterCondition(v1alpha1.ClusterConditionUpgrading)
+				//check the timestamps
 				Ω(condition.LastUpdateTime).NotTo(Equal(""))
 				Ω(condition.LastTransitionTime).NotTo(Equal(""))
 			})
 		})
 	})
+	Context("set pods Error condition  upgrade failed to be true", func() {
+		BeforeEach(func() {
+			p.Status.SetErrorConditionFalse()
+			p.Status.SetErrorConditionTrue("UpgradeFailed", " ")
+		})
+		It("should have pods Error condition with true status using function", func() {
+			Ω(p.Status.IsClusterInUpgradeFailedState()).To(Equal(true))
+		})
+		It("should have pods Error condition with true status", func() {
+			_, condition := p.Status.GetClusterCondition(v1alpha1.ClusterConditionError)
+			Ω(condition.Status).To(Equal(corev1.ConditionTrue))
+
+		})
+		It("Checking ClusterInUpgradeFailedOrRollbackState and It should return true", func() {
+			Ω(p.Status.IsClusterInUpgradeFailedOrRollbackState()).To(Equal(true))
+		})
+
+	})
+
+	Context("set pods Error condition  rollback failed to be true", func() {
+		BeforeEach(func() {
+			p.Status.SetErrorConditionFalse()
+			p.Status.SetErrorConditionTrue("RollbackFailed", " ")
+		})
+		It("should return rollback failed state to true using function", func() {
+			Ω(p.Status.IsClusterInRollbackFailedState()).To(Equal(true))
+		})
+		It("should return rollback failed state to false using function", func() {
+			p.Status.SetErrorConditionTrue("some err", "")
+			Ω(p.Status.IsClusterInRollbackFailedState()).To(Equal(false))
+		})
+
+		It("should have pods Error condition with true status using function", func() {
+			Ω(p.Status.IsClusterInErrorState()).To(Equal(true))
+		})
+		It("should have pods Error condition with true status", func() {
+			_, condition := p.Status.GetClusterCondition(v1alpha1.ClusterConditionError)
+			Ω(condition.Status).To(Equal(corev1.ConditionTrue))
+			Ω(condition.Message).To(Equal(" "))
+			Ω(condition.Reason).To(Equal("RollbackFailed"))
```

Comment:
The Reason is being set to "some err" and assertions is checking for `Reason` to be `RollbackFailed` ?

Reply:
we are setting reason as `RollbackFailed` at line no:234 in BeforeEach function

Reply:
> we are setting reason as `RollbackFailed` at line no:234 in BeforeEach function

I have changed the order of tests now.